IdentityCore/src/MSIDJsonSerializableFactory.m

```diff
@@ -45,11 +46,35 @@ + (void)registerClass:(Class<MSIDJsonSerializable>)aClass forClassType:(NSString
     }
 }
 
++ (void)mapJSONKey:(NSString *)key
+          keyValue:(NSString *)keyValue
+       kindOfClass:(Class)aClass
+       toClassType:(NSString *)classType
+{
+    if (!key || !keyValue || !aClass || !classType) return;
+    if (![key isKindOfClass:NSString.class]) return;
+    if (![keyValue isKindOfClass:NSString.class]) return;
+    if (![aClass conformsToProtocol:@protocol(MSIDJsonSerializable)]) return;
+    if (![classType isKindOfClass:NSString.class]) return;
+    
+    @synchronized(self)
+    {
+        static dispatch_once_t once;
+        dispatch_once(&once, ^{
+            s_keysMap = [NSMutableDictionary new];
```

Comment:
do we need two different containers in this class? Both seem to map class to a string, so we could reuse.

Reply:
Agreed to keep both.